libs/core/langchain_core/messages/utils.py

```diff
@@ -845,6 +909,610 @@ def list_token_counter(messages: Sequence[BaseMessage]) -> int:
         )
 
 
+@_runnable_support(supports_single=True)
+def format_messages_as(
+    messages: Union[MessageLikeRepresentation, Iterable[MessageLikeRepresentation]],
+    *,
+    format: Literal["openai", "anthropic"],
+    text: Literal["string", "block"],
+) -> Union[BaseMessage, List[BaseMessage]]:
+    """Convert message contents into a standard format.
+
+    .. versionadded:: 0.2.36
+
+    Args:
+        messages: Message-like object or iterable of objects whose contents are already
+            in OpenAI, Anthropic, Bedrock Converse, or VertexAI formats.
+        format: Format to convert message contents to.
+        text: How to format text contents. If ``text='string'``  then any string
```

Comment:
could we do `content_format` or something instead of `text`?

Reply:
and the above could be "provider" or something

Reply:
content_format feels a little too broad given it only relates to text blocks — text_format?

Reply:
sure!